File: Lab 5/TCylinder.py
from TFigure import TFigure
import math

class TCylinder(TFigure):
    def __init__(self, height, radius):
        super().__init__(height)
        self.__radius = radius

    @property
    def radius(self):
        return self.__radius

    @radius.setter
    def radius(self, radius):
        self.__radius = radius

    def GetS(self)-> float:
        return self.GetSOs() * 2 + 2 * math.pi * self.__radius * self._height

    def GetSOs(self)-> float:
        return self.__radius * self.__radius

    def GetV(self)-> float:
        return self.GetSOs() * self._height

File: Lab 5/TCylinder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_5
{
    internal class TCylinder : TFigure,ICloneable
    {
        double radius;
        public double Radius { get { return radius; } set { radius = value; } }
        public TCylinder(double height, double radius) : base(height)
        {
            this.radius = radius;
        }
        public override double GetS()
        {
            return GetSOS() * 2 + 2 * Math.PI * Radius * height;
        }

        public override double GetV()
        {
            return GetSOS() * height;
        }

        protected override double GetSOS()
        {
            return Radius * Radius * Math.PI;
        }

        public object Clone()
        {
            return new TCylinder(this.height,this.radius);
        }
    }
}
